dask_cuda/initialize.py

```diff
@@ -25,98 +64,76 @@ def _create_cuda_context_handler():
-        )
-        ctx = has_cuda_context()
-        if (
-            ctx.has_context
-            and not distributed.comm.ucx.cuda_context_created.has_context
-        ):
-            distributed.comm.ucx._warn_existing_cuda_context(ctx, os.getpid())
-
-        _create_cuda_context_handler()
-
-        if (
-            distributed.comm.ucx.cuda_context_created is False
-            or distributed.comm.ucx.cuda_context_created.has_context
-        ):
-            ctx = has_cuda_context()
-            if ctx.has_context and ctx.device_info != cuda_visible_device:
-                distributed.comm.ucx._warn_cuda_context_wrong_device(
-                    cuda_visible_device, ctx.device_info, os.getpid()
-                )
+    Warns if a pre-existing CUDA context already existed or if the resulting CUDA
+    context was created in the wrong device.
 
-    except Exception:
-        logger.error("Unable to start CUDA Context", exc_info=True)
+    This function is almost an identical duplicate from
+    `distributed_ucxx.ucxx.init_once`, the duplication is necessary because Dask-CUDA
+    needs to support `protocol="tcp"` as well, even when distributed-ucxx is not
+    installed, but this here runs _after_ comms have started, which is fine for TCP
+    because the time when CUDA context is created is not important. The code needs to
+    live also in distributed-ucxx because there the time when a CUDA context is created
+    matters, and it needs to happen _before_ UCX is initialized, but comms in
+    Distributed is initialized before preload, and thus only after this function
+    executes.
 
+    Raises
+    ------
+    Exception
+        If anything wrong happened during context initialization.
 
-def _initialize_ucx():
-    try:
-        import distributed.comm.ucx
+    Returns
+    -------
+    None
+    """
+    cuda_visible_device = get_device_index_and_uuid(
+        os.environ.get("CUDA_VISIBLE_DEVICES", "0").split(",")[0]
+    )
+    pre_existing_cuda_context = has_cuda_context()
+    if pre_existing_cuda_context.has_context:
+        _warn_existing_cuda_context(pre_existing_cuda_context.device_info, os.getpid())
+
+    _create_cuda_context_handler()
+
+    cuda_context_created = has_cuda_context()
+    if (
+        cuda_context_created.has_context
+        and cuda_context_created.device_info.uuid != cuda_visible_device.uuid
+    ):
+        _warn_cuda_context_wrong_device(
+            cuda_visible_device, cuda_context_created.device_info, os.getpid()
+        )
 
-        distributed.comm.ucx.init_once()
-    except ModuleNotFoundError:
-        # UCX initialization has to be delegated to Distributed, it will take care
-        # of setting correct environment variables and importing `ucp` after that.
-        # Therefore if ``import ucp`` fails we can just continue here.
-        pass
 
+def _create_cuda_context():
+    global pre_existing_cuda_context, cuda_context_created
```

Comment:
I could be misreading the diff, but is this `global` doing anyting? I dont' see `pre_existing_cuda_context` or `cuda_context_created` reassigned here.

Should the `global` move to `_create_cuda_context_and_warn` instead, where those are assigned?